gradle/dependencies.gradle

```diff
@@ -15,367 +15,386 @@
-                }
-            }
-            withModule('org.apache.maven.wagon:wagon-http-shared4') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group == 'org.jsoup' || it.group == 'commons-logging' || it.group == 'org.apache.httpcomponents' }}
-                }
-                withVariant('runtime') {
-                    withDependencies { it.removeAll { it.group == 'org.jsoup' || it.group == 'commons-logging' || it.group == 'org.apache.httpcomponents' }}
-                }
-            }
-            withModule('net.sourceforge.nekohtml:nekohtml') {
-                withVariant('default') {
-                    withDependencies {
-                        it.findAll { it.group == 'xerces' }.each {
-                            changeVersionInMetadata it, '2.11.0'
-                        }
-                    }
-                }
-            }
-            withModule('org.eclipse.jgit:org.eclipse.jgit') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group == 'com.googlecode.javaewah' }}
-                }
-            }
-            withModule('com.amazonaws:aws-java-sdk-core') {
-                withVariant('default') {
-                    withDependencies {
-                        it.removeAll { it.name == 'jackson-dataformat-cbor' || it.group == 'commons-logging' }
-                        it.findAll { it.name == 'httpclient' }.each { changeVersionInMetadata it, versions.httpclient }
-                    }
-                }
-                withVariant('runtime') {
-                    withDependencies {
-                        it.removeAll { it.name == 'jackson-dataformat-cbor' || it.group == 'commons-logging' }
-                        it.findAll { it.name == 'httpclient' }.each { changeVersionInMetadata it, versions.httpclient }
-                    }
+                allVariants {
+                    withDependencies { it.removeAll { it.group == 'org.sonatype.sisu' } }
                 }
             }
-            withModule('dom4j:dom4j') {
-                withVariant('runtime') {
-                    withDependencies { it.removeAll { it.group == 'xml-apis' }}
-                }
-            }
-            withModule('com.google.code.findbugs:findbugs') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group != 'dom4j' }}
+
+            // Gradle distribution - replace similar library with different coordinates
+            replaceGoogleCollectionsWithGuava(handler, 'org.codehaus.plexus:plexus-container-default')
+
+            replaceJunitDepWithJunit(handler, 'org.jmock:jmock-junit4')
+
+            replaceBeanshellWithApacheBeanshell(handler, 'org.testng:testng')
+
+            replaceLog4JWithAdapter(handler, 'org.apache.xbean:xbean-reflect')
+
+            replaceJCLWithAdapter(handler, 'org.apache.xbean:xbean-reflect')
```

Comment:
All of these `replaceFooWithBar` rules will give us a nice concrete use case for 'replace/provides constraints'